### PR TITLE
Promise

### PR DESCRIPTION
**Pull request by lyw0149**
Solve `callback hell` using promise.

http://www.summa.com/blog/avoiding-callback-hell-while-using-mongoose

@ethkim  @nlpenguin @djflexible @ifndefdeadmau5 @MechanicKim

### CODE REVIEW

**Comment by ethkim** [comment]
PR looks harmless though I have very little visibility on Promise stuff. @ifndefdeadmau5 : could you spare an extra eye? 

Also, make sure you sync since [PR26](https://github.com/sv-bootcamp/yoda/pull/26) is also pending.

File: src/server/models/match.model.js [hunk 1]
```diff
@@ -15,7 +15,7 @@ let matchSchema = new Schema({
     default: Date.now(),
   },
   response_date: Date,
-  //status of match {pending:0, accepted:1, rejected:2}
+  //status of match {pending:2, accepted:1, rejected:0}
```
Comment (lyw0149) [inline src/server/models/match.model.js]: This mistake made only at comment. All code is based on 
{pending:2, accepted:1, rejected:0}

**Comment by ifndefdeadmau5** [comment]
It seems work correctly to me too, and also enhanced readability. So LGTM!

**Comment by ifndefdeadmau5** [comment]
LGTM

**Comment by ifndefdeadmau5** [comment]
👍

**Pull request merged**